ngraph/frontend/paddlepaddle/src/op/interp.cpp

```diff
@@ -33,30 +85,107 @@ NamedOutputs nearest_interp_v2 (const NodeContext& node) {
     auto out_h = node.get_attribute<int>("out_h");
     auto scale = node.get_attribute<std::vector<float>>("scale");
-    if (out_w <= 0 || out_h <= 0) {
+
+    Output<Node> scales;
+    Output<Node> target_spatial_shape;
+    //TODO reduce duplicate codes
+    if(node.has_ng_input("OutSize"))
+    {
+        attrs.shape_calculation_mode = ShapeCalcMode::sizes;
+        auto hw_shape = node.get_ng_input("OutSize");
+        const auto shape_of_x = std::make_shared<opset6::ShapeOf>(x);
+        auto shape_begin = opset6::Constant::create(element::Type_t::i64, {1}, {0});
+        auto shape_end = opset6::Constant::create(element::Type_t::i32, Shape{1}, {2});
+        auto nc_node = std::make_shared<opset6::StridedSlice>(shape_of_x, shape_begin, shape_end, std::vector<int64_t>{0}, std::vector<int64_t>{0});
+        target_spatial_shape = std::make_shared<opset6::Concat>(OutputVector{nc_node, hw_shape}, 0);
+        scales = calculate_scales_based_on_sizes(x, target_spatial_shape);
+    }
+    else if (out_w <= 0 || out_h <= 0) {
         attrs.shape_calculation_mode = ShapeCalcMode::scales;
+        target_spatial_shape = calculate_output_shape_based_on_scales(x, scale, scales);
     }
     else {
         attrs.shape_calculation_mode = ShapeCalcMode::sizes;
+        target_spatial_shape = extract_out_sizes(x, {out_h, out_w});
+        scales = calculate_scales_based_on_sizes(x, target_spatial_shape);
+    }
+
+    attrs.coordinate_transformation_mode = CoordinateTransformMode::asymmetric;
+    attrs.nearest_mode = Nearest_mode::round_prefer_floor;
+    attrs.antialias = false;
+    attrs.pads_begin = {0, 0, 0, 0};
+    attrs.pads_end = {0, 0, 0, 0};
+
+    return node.default_single_output_mapping({std::make_shared<ngraph::opset6::Interpolate>(x, target_spatial_shape, scales, attrs)}, {"Out"});
+}
+
+NamedOutputs bilinear_interp_v2 (const NodeContext& node) {
+    auto x = node.get_ng_input("X");
+
+    using InterpolateMode = ngraph::opset6::Interpolate::InterpolateMode;
+    using CoordinateTransformMode = ngraph::opset6::Interpolate::CoordinateTransformMode;
+    using Nearest_mode = ngraph::opset6::Interpolate::NearestMode;
+    using InterpolateAttrs = ngraph::opset6::Interpolate::InterpolateAttrs;
+    using ShapeCalcMode = ngraph::opset6::Interpolate::ShapeCalcMode;
+
+    InterpolateAttrs attrs;
+
+    attrs.mode = InterpolateMode::linear_onnx;
+
+    auto out_w = node.get_attribute<int>("out_w");
+    auto out_h = node.get_attribute<int>("out_h");
+    auto scale = node.get_attribute<std::vector<float>>("scale");
+    Output<Node> scales;
+    Output<Node> target_spatial_shape;
+    //TODO reduce duplicate codes
+    if(node.has_ng_input("OutSize"))
+    {
+        attrs.shape_calculation_mode = ShapeCalcMode::sizes;
+        auto hw_shape = node.get_ng_input("OutSize");
+        const auto shape_of_x = std::make_shared<opset6::ShapeOf>(x);
+        auto shape_begin = opset6::Constant::create(element::Type_t::i64, {1}, {0});
```

Comment:
can be just element::i64, not element::Type_t::i64

Reply:
remove this in 2813bc5246dc6793f64111aeddebbb8eee9b0fcd